core/src/main/java/org/infinispan/remoting/transport/AggregateBackupResponse.java

```diff
@@ -1,112 +1,95 @@
-      for (BackupResponse br : responses) {
-         br.waitForBackupToFinish();
-      }
-   }
+   private final BackupResponse one;
+   private final BackupResponse two;
 
-   @Override
-   public Map<String, Throwable> getFailedBackups() {
-      Map<String, Throwable> result = new HashMap<>();
-      for (BackupResponse br : responses) {
-         result.putAll(br.getFailedBackups());
-      }
-      return result;
-   }
-
-   @Override
-   public Set<String> getCommunicationErrors() {
-      Set<String> result = new HashSet<>();
-      for (BackupResponse br : responses) {
-         result.addAll(br.getCommunicationErrors());
-      }
-      return result;
+   private AggregateBackupResponse(BackupResponse one, BackupResponse two) {
+      this.one = one;
+      this.two = two;
    }
 
    @Override
    public long getSendTimeMillis() {
-      long min = Long.MAX_VALUE;
-      for (BackupResponse br: responses) {
-         min = Math.min(br.getSendTimeMillis(), min);
-      }
-      return min;
+      return Math.min(one.getSendTimeMillis(), two.getSendTimeMillis());
    }
 
    @Override
    public String toString() {
       return "AggregateBackupResponse{" +
-            "responses=" + responses +
+            "responses=" + one + ", " + two +
             '}';
    }
 
    @Override
    public boolean equals(Object o) {
-      if (this == o) return true;
-      if (!(o instanceof AggregateBackupResponse)) return false;
+      if (this == o) {
+         return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+         return false;
+      }
 
       AggregateBackupResponse that = (AggregateBackupResponse) o;
-
-      return responses != null ? responses.equals(that.responses) : that.responses == null;
+      return one.equals(that.one) && two.equals(that.two);
    }
 
    @Override
    public int hashCode() {
-      return responses != null ? responses.hashCode() : 0;
+      int result = one.hashCode();
+      result = 31 * result + two.hashCode();
+      return result;
    }
 
    @Override
-   public boolean isEmpty() {
-      for (BackupResponse br : responses) {
-         if (!br.isEmpty()) return false;
-      }
-      return true;
+   public void notifyFinish(Consumer<Long> timeElapsedConsumer) {
+      one.notifyFinish(timeElapsedConsumer);
+      two.notifyFinish(timeElapsedConsumer);
    }
 
    @Override
-   public void notifyFinish(LongConsumer timeElapsedConsumer) {
-      for (BackupResponse br : responses) {
-         br.notifyFinish(timeElapsedConsumer);
-      }
+   public InvocationStage toInvocationStage(XSiteResponseHandler handler) {
+      //kind of messy...
+      CompletableFuture<Object> oneFuture = one.toInvocationStage(handler).toCompletableFuture();
+      CompletableFuture<Object> twoFuture = two.toInvocationStage(handler).toCompletableFuture();
+      return new SimpleAsyncInvocationStage(CompletableFuture.allOf(oneFuture, twoFuture));
```

Comment:
the only use case for this class is when you have a tx cache with 2 sites configured, 1 async and another sync. 
you need to send a `PrepareCommand(onePhase=true)` for the async site and the `CommitCommand` to the second site. 

I can add `invokeXSiteCommand(Map<XSiteBackup, XSiteReplicableCommand>)` but I didn't want to add too much to this PR (it is already large enough) 
I'm open to suggestion :)

Reply:
I don't want `invokeXSiteCommand(Map<XSiteBackup, XSiteReplicableCommand>)`, just `invokeXSiteCommand(XSiteBackup, XSiteReplicableCommand)`.

`BackupSenderImpl` would invoke the command multiple times, process the responses (possibly with `CompletionStages.aggregateCompletionStage()`), and return a single `CompletionStage`.

I was thinking it would actually reduce the size of the PR, because you wouldn't need the aggregate responses and all the `InvocationStage` conversions, but I'll leave it up to you.

Reply:
the `invokeXSite()` returns a `BackupResponse`. How is that different than using the `AggregateResponse`?

Reply:
I don't understand the question. If RpcManagerImpl and JGroupsTransport only had a method `invokeXSiteCommand(XSiteBackup, XSiteReplicableCommand) -> CompletionStage<Void>` then you wouldn't need either `AggregateBackupResponse` or `BackupResponse`.